internal/core/operations/executor.go

```diff
@@ -25,19 +25,21 @@ package operations
 import (
 	"context"
 
+	"go.uber.org/zap"
+
 	"github.com/topfreegames/maestro/internal/core/entities/operation"
 )
 
 // Executor is where the actual operation logic is implemented, and it will
 // receive as input its correlated definition.
 type Executor interface {
-	// This is where the operation logic will live; it will receive a context
+	// Execute is where the operation logic will live; it will receive a context
 	// that will be used for deadline and cancelation. This function has only
 	// one return which is the operation error (if any);
-	Execute(ctx context.Context, op *operation.Operation, definition Definition) error
-	// This function is called if Execute returns an error. This will be used
+	Execute(ctx context.Context, op *operation.Operation, definition Definition, logger *zap.Logger) error
```

Comment:
Any specific reason for adding the logger as an argument on the `Executor`? In other places, we're following the pattern of using the "static" logger instance instead of injecting it (like [here](https://github.com/topfreegames/maestro/blob/next/internal/core/workers/operation_execution_worker/operation_execution_worker.go#L60), for example).

Reply:
The idea here is to have the same metadata assumed on the OperationExecutionWorker: scheduler name, operation id, and so on...  But I'm not comfortable having it on the interface...

Reply:
@gabrielcorado, @luizmiranda7 described the initial idea that I had, but seeing it again we already have all of that info inside the executor and we wouldn't need to receive the logger as an argument. For preserving the initial interface definition, I prefer to use the "static" logger